lib/fastifySession.js

```diff
@@ -120,7 +121,16 @@ function onRequest (options) {
     if (!sessionId) {
       newSession(secret, request, cookieOpts, idGenerator, done)
     } else {
-      decryptSession(sessionId, options, request, done)
+      let sessionToDecrypt = sessionId
+
+      if (unsignSignedCookie) {
+        const unsignedCookie = reply.unsignCookie(sessionId)
+        if (unsignedCookie.valid) {
+          sessionToDecrypt = unsignedCookie.value
+        }
```

Comment:
what if the unsigned cookie is not valid? do you think it's sensible to ignore it and proceed?

Reply:
It's the same practical behavior we had before. What'll happen is that the `decryptSession` will be unable to decrypt the cookie and will, in turn, create a new session for the user. In case we can't unsign it I don't think there's much we can do but that. I've simply removed some bloat from this snippet.

Another approach that came to mind was to remove the `unsignSignedCookie` flag altogether and **always** check if the cookie is signed. In code it'd only remove the `if (unsignedCookie) {` check from around this block and we'd run the `reply.unsignCookie` on every request. Thinking of the library I think this would be a neat solution, given we'd be adding support for that feature transparently to the user, no need to tweak new flags. On the other hand, on my machine benchmarks were about 10% slower then (from ~273 ops/sec to ~254 ops/sec). Given that performance regression I decided to simply keep the approach that had been previously approved.

Reply:
thanks for the explanation! I'm happy to keep it as it is, just wanted to make sure we're intentional about it